lib/modules/manager/fleet/extract.spec.ts

```diff
@@ -0,0 +1,149 @@
+const inValidFleetYaml = Fixtures.get('invalid_fleet.yaml');
+
+const validGitRepoYaml = Fixtures.get('valid_gitrepo.yaml');
+const invalidGitRepoYaml = Fixtures.get('invalid_gitrepo.yaml');
+
+const configMapYaml = Fixtures.get('configmap.yaml', '../kubernetes');
+
+describe('modules/manager/fleet/extract', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('extractPackageFile()', () => {
+    it('should return null if empty content', () => {
+      const result = extractPackageFile('', 'fleet.yaml');
+
+      expect(result).toBeNull();
+    });
+
+    it('should return null if a unknown manifest is supplied', () => {
+      const result = extractPackageFile(configMapYaml, 'fleet.yaml');
+
+      expect(result).toBeNull();
+    });
+
+    describe('fleet.yaml', () => {
+      it('should return null if content is a malformed YAML', () => {
+        jest.spyOn(yaml, 'loadAll').mockImplementation(() => {
+          throw new Error();
+        });
+        const result = extractPackageFile('test-', 'fleet.yaml');
+
+        expect(result).toBeNull();
+      });
+
+      it('should parse valid configuration', () => {
+        const result = extractPackageFile(validFleetYaml, 'fleet.yaml');
+
+        expect(result).not.toBeNull();
+        expect(result?.deps).toMatchObject([
+          {
+            currentValue: 'v1.8.0',
+            datasource: 'helm',
+            depName: 'cert-manager',
+            registryUrls: ['https://charts.jetstack.io'],
+            depType: 'fleet',
+          },
+          {
+            currentValue: '3.17.7',
+            datasource: 'helm',
+            depName: 'logging-operator',
+            registryUrls: ['https://kubernetes-charts.banzaicloud.com'],
+            depType: 'fleet',
+          },
+        ]);
+      });
+
+      it('should parse parse invalid configurations', () => {
+        const result = extractPackageFile(inValidFleetYaml, 'fleet.yaml');
+
+        expect(result).not.toBeNull();
+        expect(result?.deps).toMatchObject([
+          {
+            skipReason: 'no-version',
+            datasource: 'helm',
+            depName: 'cert-manager',
+            registryUrls: ['https://charts.jetstack.io'],
+            depType: 'fleet',
+          },
+          {
+            datasource: 'helm',
+            depName: 'logging-operator',
+            skipReason: 'no-repository',
+            depType: 'fleet',
+          },
+          {
+            datasource: 'helm',
+            skipReason: 'missing-depname',
+            depType: 'fleet',
+          },
+          {
+            datasource: 'helm',
+            depName: './charts/example',
+            skipReason: 'local-chart',
+            depType: 'fleet',
+          },
+        ]);
+      });
+    });
+
+    describe('GitRepo', () => {
+      it('should return null if content is a malformed YAML', () => {
+        jest.spyOn(yaml, 'loadAll').mockImplementation(() => {
+          throw new Error();
+        });
```

Comment:
Do we reall y need this? didn't it already throw for `test` input?

Reply:
https://github.com/renovatebot/renovate/blob/56eb0bae9aa8d850f6d45981188c44de6e10fc23/lib/modules/manager/kubernetes/extract.spec.ts#L81-L87

Reply:
No, it hasn't thrown an exception with this input.    
I have also tried single space indentation and multiple keys in one line, all has been parsed.    

Anyway the smaller-than sign seems to trigger it.